src/main/java/ch/uzh/ifi/hase/soprafs24/controller/TeamController.java

```diff
@@ -144,6 +145,38 @@ public List<TaskGetDTO> getTasks(
         .collect(Collectors.toList());
   }
 
+  /*
+   * GET method which uses getTasksByTeamIdAndStatus method from TaskService to get tasks by status
+   */
+  @GetMapping(value = "/teams/{ID}/tasks", params = {"status"})
+  @ResponseBody
+  public List<TaskGetDTO> getTasksByStatus(@PathVariable("ID") Long teamId,
+      @RequestHeader("Authorization") String token, @RequestParam("status") List<String> status) {
+    // auth user
+    authorizationService.isAuthorizedAndBelongsToTeam(token, teamId);
+
+    // convert status to TaskStatus
+    List<TaskStatus> taskStatusList;
+    try {
+      taskStatusList = status.stream().map(TaskStatus::valueOf).collect(Collectors.toList());
+    } catch (IllegalArgumentException ex) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid task status.", ex);
+    }
+
+    // get tasks
+    List<Task> tasks = taskService.getTasksByTeamIdAndStatus(teamId, taskStatusList);
+
+    // if tasks list is empty, return empty list
+    if (tasks.isEmpty()) {
+      return new ArrayList<>();
+    }
+
+    // convert internal representation of tasks back to API
+    return tasks.stream()
+        .map(DTOMapper.INSTANCE::convertEntityToTaskGetDTO)
+        .collect(Collectors.toList());
```

Comment:
shouldn't empty lists be handled automatically? I.e., why check for `isEmtpy()` needed?

Reply:
I don't know how to handle it automatically @5v3nn this was my way of implementation do I have to change it?

Reply:
ok if not possible